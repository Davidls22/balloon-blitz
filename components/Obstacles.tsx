import React from 'react';
import { View, Image } from 'react-native';

type Props = {
  left: number;
  bottom: number;
  width: number;
  height: number;
  image: any; 
};

export default function Obstacles({ left, bottom, width, height, image }: Props) {
  if (!image) return null;

  return (
    <View
      style={{
        position: 'absolute',
        width,
        height,
        left,
        bottom,
      }}
    >
      <Image
        source={image}
        style={{ width: '100%', height: '100%' }}
        resizeMode="stretch"
      />
    </View>
  );
}