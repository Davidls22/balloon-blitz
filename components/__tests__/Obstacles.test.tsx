import React from 'react';
import renderer from 'react-test-renderer';
import Obstacles from '../Obstacles';

test('renders obstacle', () => {
  const mockImage = { uri: 'mockImage' };
  const tree = renderer
    .create(<Obstacles left={50} bottom={100} width={100} height={200} image={mockImage} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders null if no image', () => {
  const tree = renderer
    .create(<Obstacles left={50} bottom={100} width={100} height={200} image={null} />)
    .toJSON();
  expect(tree).toBeNull();
});