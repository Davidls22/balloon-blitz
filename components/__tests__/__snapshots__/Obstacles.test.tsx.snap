// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders obstacle 1`] = `
<View
  style={
    {
      "bottom": 100,
      "height": 200,
      "left": 50,
      "position": "absolute",
      "width": 100,
    }
  }
>
  <Image
    resizeMode="stretch"
    source={
      {
        "uri": "mockImage",
      }
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
